From 64093965168dfc55dad938db36ed487d9e72a32f Mon Sep 17 00:00:00 2001
From: hengluo <heng.luo@intel.com>
Date: Wed, 17 Oct 2018 13:34:56 +0800
Subject: [PATCH] SystemUpdater: allow to access removable storage

Add WRITE_MEDIA_STORAGE permission in SystemUpdater,
allow it to select OTA package in removable storage,
such as USB stick or SD card.

Change-Id: I33902661e728ea5763d2b601ac2109e7370acb10
Tracked-On:
Signed-off-by: hengluo <heng.luo@intel.com>
---

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 1a77c05..3dbe8dc 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -19,6 +19,7 @@
         xmlns:android="http://schemas.android.com/apk/res/android"
         package="com.android.car.systemupdater">
 
+    <uses-permission android:name="android.permission.WRITE_MEDIA_STORAGE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.REBOOT" />
diff --git a/src/com/android/car/systemupdater/SystemUpdaterActivity.java b/src/com/android/car/systemupdater/SystemUpdaterActivity.java
index c53ea18..ee1c1c8 100644
--- a/src/com/android/car/systemupdater/SystemUpdaterActivity.java
+++ b/src/com/android/car/systemupdater/SystemUpdaterActivity.java
@@ -39,7 +39,8 @@
     private static final int STORAGE_PERMISSIONS_REQUEST_CODE = 0;
     private static final String[] REQUIRED_STORAGE_PERMISSIONS = new String[]{
             Manifest.permission.READ_EXTERNAL_STORAGE,
-            Manifest.permission.WRITE_EXTERNAL_STORAGE
+            Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.WRITE_MEDIA_STORAGE
     };
 
     @Override
